docs/script_matching.md

```diff
@@ -0,0 +1,93 @@
+One central task for skribo is choosing a font most suitable for the requested script. The requested script is a combination of an explicit lang tag and locale system settings.
+
+Part of what makes this requirement tricky is that there is a fuzzy line between the scope of skribo and the scope of [font-kit]. Conceptually, skribo should be a consumer of font metadata, and font-kit is a provider. At a finer grain, there are decisions about exactly what that interface should look like. The main purpose of this document is to describe the problems to be solved.
+
+## Background
+
+Selection of font is not strictly based on Unicode coverage, but is also affected by locale. The biggest impact is on Han unification, but there are other examples. Suitability of a font for a script is *not* encoded in the font's tables, but is metadata about the system's font configuration. Mechanisms for accessing this metadata vary widely from platform to platform.
+
+A significant amount of research for this topic was reading through the [Skia] code, as that's the platform abstraction used by Chromium, so can be used as a rough first draft of requirements for correct cross-platform text layout for the Web. I also used Gecko and Qt as sources.
+
+The main focus of this document is font fallback, but a related platform-specific concern is resolution of generic aliases (system UI font, generic serif, etc). This may be out of scope for skribo itself, but likely within the scope of font-kit. More clarity on requirements for each crate is needed.
+
+TODO: write a clear requirement, link to this section of requirements doc.
+
+## BCP-47
+
+I strongly recommend the use of BCP-47 as the identifier for language, script, and other locale metadata. This is an easy decision for web use cases, as it is the standard for the [lang] tag. The main challenge is that mechanisms for system font metadata in general predate BCP-47, so there will be some impedance matching.
+
+TODO: investigate Rust ecosystem for common BCP-47 tag representation.
```

Comment:
cc @zbraniecki

Pretty sure fluent-rs needs this too, worth knowing what y'all are using

Reply:
Yeah, I'm curious what work is happening. I did just a bit of searching, didn't find any clear consensus on what people are doing, so would be very happy to hear from any efforts in this space.

Reply:
I'm using a crate [fluent-locale](https://crates.io/crates/fluent-locale) which provides the basic BCP47 locale management and negotiation.

I'm hoping for a `Locale` class to be added to `unic`, but that is going a bit slower and awaits https://github.com/open-i18n/rust-unic/issues/195#issuecomment-460841147